### PR TITLE
Add OpenTracing Python implementation

### PR DESCRIPTION
**Pull request by bg451**
Now that https://github.com/opentracing/basictracer-python is a thing, I've added a python opentracing implementation for Python. What's really cool about this is that it's 100% compatible with the Golang implementation, so two processes can send each other spans and join traces using the inject/join methods.

I'm not too sure how python dependencies work(I'm not too great with python), but I changed setup.py to use setuptools to bring in the basictracer dependency.

### CODE REVIEW

**Comment by bg451** [comment]
ping @slimsag

File: python/setup.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
+# from distutils.core import setup
+from setuptools import setup
```
Comment (slimsag) [inline python/setup.py]: Should just remove distutils import instead of leaving it as a comment?

**Comment by slimsag** [comment]
This looks like a great improvement, thanks! I have one minor comment inline and will merge after it's addressed.

**Comment by bg451** [comment]
Addressed your comment, should be good to go! @slimsag

**Comment by slimsag** [comment]
Thanks Brandon! 😃

**Pull request merged**